### PR TITLE
Improve coverage of tests in RouterCommon.sol

### LINKED ISSUE #715: Coverage Report - Test RouterCommon.sol
Test the following functions

- [x] _sendTokenOut (When wethIsEth and tokenOut is _weth)
- [x] _takeTokenIn (When balance < amountIn)
- [x] constructor (check that vault, weth and permit2 are correctly assigned)

### PR DESCRIPTION
**Pull request by joaobrunoah**
# Description

Improve test coverage for _takeTokenIn and _sendTokenOut functions, which were not covered for eth transactions.

## Type of change

- [ ] Bug fix 
- [ ] New feature 
- [ ] Breaking change 
- [ ] Dependency changes
- [x] Code refactor / cleanup
- [ ] Documentation or wording changes
- [ ] Other

## Checklist:

- [x] The diff is legible and has no extraneous changes
- [x] Complex code has been commented, including external interfaces
- [x] Tests have 100% code coverage
- [x] The base branch is either `main`, or there's a description of how to merge

## Issue Resolution

Closes #715

### CODE REVIEW

**Review by EndymionJkb — COMMENTED** [review]
Looks good; just a couple comments. Also, what should I look at to see the difference? `yarn coverage` seems to show the same numbers.

**Comment by joaobrunoah** [comment]
> Looks good; just a couple comments. Also, what should I look at to see the difference? `yarn coverage` seems to show the same numbers.

Cool, I just adjusted the tests. About coverage, I'm using only forge coverage for now. To get the html report, I'm iusing the following coverage.sh:

```
#!/bin/bash

set -e # exit on error

# reduces the amount of tests in fuzzing, so coverage runs faster
export FOUNDRY_PROFILE=coverage
export CURRENT_PACKAGE=$(basename "$PWD")

# generates lcov.info
forge coverage --report lcov

# Initialize variables
current_file=""
lines_found=0
lines_hit=0

sed "s/\/.*$CURRENT_PACKAGE.//g" lcov.info > lcov-clearfolders.info
sed -i -e "s/\.\.contracts\//contracts\//g" lcov-clearfolders.info

# generates coverage/lcov.info
# COVERAGE=true yarn hardhat coverage

# Foundry uses relative paths but Hardhat uses absolute paths.
# Convert absolute paths to relative paths for consistency.
# sed -i -e "s/\/.*$CURRENT_PACKAGE.//g" coverage/lcov.info

# Merge lcov files
# --add-tracefile coverage/lcov.info \
lcov \
  --rc lcov_branch_coverage=1 \
  --rc derive_function_end_line=0 \
  --add-tracefile lcov-clearfolders.info \
  --output-file lcov-merged.info \
  --no-checksum

if [$CURRENT_PACKAGE = 'vault']; then
  # Filter out node_modules, test, and mock files
  lcov \
    --rc lcov_branch_coverage=1 \
    --rc derive_function_end_line=0 \
    --remove lcov-merged.info \
    "*node_modules*" "*test*" "*mock*" \
    --output-file lcov-filtered.info
else
  # Filter out node_modules, test, mock files and vault contracts
  lcov \
    --rc lcov_branch_coverage=1 \
    --rc derive_function_end_line=0 \
    --remove lcov-merged.info \
    "*node_modules*" "*test*" "*mock*" "*/vault/*"\
    --output-file lcov-filtered.info
fi

# Generate summary
lcov \
  --rc lcov_branch_coverage=1 \
  --rc derive_function_end_line=0 \
  --list lcov-filtered.info

# Open more granular breakdown in browser
rm -rf coverage-genhtml/
genhtml \
  --rc genhtml_branch_coverage=1 \
  --rc derive_function_end_line=0 \
  --output-directory coverage-genhtml \
  lcov-filtered.info
open coverage-genhtml/index.html

# Delete temp files
rm -rf lcov-*.info lcov-*.info-e coverage/ lcov.info coverage.json
```

File: pkg/vault/test/foundry/RouterCommon.t.sol [hunk 1]
```diff
@@ -45,16 +53,105 @@ contract RouterCommonTest is BaseTest {
...
                     RouterCommonMock.call.selector,
-                    router,
+                    routerMock,
                     abi.encodeWithSelector(RouterCommonMock.emitSender.selector)
                 )
             )
         );
     }
+
+    function testTakeTokenInInsufficientEth() public {
```
Comment (EndymionJkb) [inline pkg/vault/test/foundry/RouterCommon.t.sol:69]:
```suggestion
    function testTakeTokenInWethIsEth() public {
```
For consistency with the rest of the tests

File: pkg/vault/test/foundry/RouterCommon.t.sol [hunk 2]
```diff
@@ -45,16 +53,105 @@ contract RouterCommonTest is BaseTest {
...
+        uint256 routerEthBalance = address(routerMock).balance;
+
+        vm.expectRevert(RouterCommon.InsufficientEth.selector);
+        routerMock.mockTakeTokenIn(bob, IERC20(weth), routerEthBalance + 1, true);
+    }
+
+    function testTakeTokenInWEthIsNotEth() public {
```
Comment (EndymionJkb) [inline pkg/vault/test/foundry/RouterCommon.t.sol:76]:
```suggestion
    function testTakeTokenInWethIsNotEth() public {
```

**Review by EndymionJkb — APPROVED** [review]
Just a couple naming nits

**Review by elshan-eth — APPROVED** [review]
LGTM!

**Pull request merged**